data/yam/agama/auto/template.libsonnet

```diff
@@ -2,22 +2,25 @@ local base_lib = import 'lib/base.libsonnet';
 local addons_lib = import 'lib/addons.libsonnet';
 local scripts_post_lib = import 'lib/scripts_post.libsonnet';
 local scripts_pre_lib = import 'lib/scripts_pre.libsonnet';
+local software_lib = import 'lib/software.var';
 local storage_lib = import 'lib/storage.libsonnet';
 
 function(addon_ha_reg_code='',
          bootloader=false,
          user=true,
          root=true,
          storage='',
+         packages='',
          patterns='',
          product='',
          registration_code='',
          scripts_pre='',
          scripts_post='') {
   [if bootloader == true then 'bootloader']: base_lib['bootloader'],
-  [if patterns != '' then 'software']: {
-    patterns: [patterns]
-  },
+  [if patterns != '' || packages != '' then 'software']: std.prune({
+    patterns: if patterns != '' then software_lib.patterns else null,
```

Comment:
else null, might not be needed if it is possible to do it like in line 26, it is creating null anyway.

Reply:
changed.